Remove case-sensitivity from language input

Recently I was trying to use forkability to analyze a project that was written in Python, such that I would recieve the Python specific suggestions.

When I was entering the project's information on the [web app](http://basicallydan.github.io/forkability/), I inputted the language for the project as "python" with the first letter lowercase and it gave the Python requirements (ex. setup.py file). However, when I entered the language as "Python" it did not give the Python requirements.

Thus it seems that the language input box is case-sensitive, such that it will not recognize programming languages if they are written with the first letter capitalized, rather than being in all lowercase.

Could this be changed so that the programming languages could be recognized despite their capitalization?

I attempted run `make compile`, however I am getting some errors when doing so.

```
browserify lib/app.js | uglifyjs -c > dist/forkability.edge.min.js
/bin/sh: browserify: command not found
/bin/sh: uglifyjs: command not found
makefile:18: recipe for target 'compile' failed
make: *** [compile] Error 127
```

I figured that the errors were just due to me having not installed `browserify` and `uglifyjs` through npm, so I installed those two modules. However after running `make compile` again I still had the same errors.

I looked into the error a bit and found that I might be able to get the compile makescript to work by putting `npm bin` in from of each of the unfound commands. So I did that and running `make compile` seemed to work. However, when I run the web app with `make serve` and attempt to enter the project information I have some odd issues.

When I go to enter the language name the button that would normally pop-up with the name of the entered language does not appear, and when I click on the "Check Forkability" button the page just refreshes.

Here is the process I am trying to be able to see the changes in the local run of the web app:

* Use Git to checkout the branch that has the changes.
* Go to the main directory of the project.
* Run `make compile`
* The following error is then outputted:
```
browserify lib/app.js | uglifyjs -c > dist/forkability.edge.min.js
/bin/sh: browserify: command not found
/bin/sh: uglifyjs: command not found
makefile:18: recipe for target 'compile' failed
make: *** [compile] Error 127
```

--- a/lib/app.js
+++ b/lib/app.js
@@ -52,12 +52,13 @@
 		};
 
 		for (var i = options.languages.length - 1; i >= 0; i--) {
-			if (languages[options.languages[i]]) {
-				lintOptions.files = merge(lintOptions.files, languages[options.languages[i]].files);
-				lintOptions.ignore = merge(lintOptions.ignore, languages[options.languages[i]].ignore);
-				supportedLanguages.push(options.languages[i]);
+			var currentLanguage = options.languages[i].toLowerCase();
+			if (languages[currentLanguage]) {
+				lintOptions.files = merge(lintOptions.files, languages[currentLanguage].files);
+				lintOptions.ignore = merge(lintOptions.ignore, languages[currentLanguage].ignore);
+				supportedLanguages.push(currentLanguage);
 			} else {
-				console.warn('Sorry, Forkability doesn\'t support the language', options.languages[i] + '. If you\'re keen you can open a pull request at https://github.com/basicallydan/forkability/issues');
+				console.warn('Sorry, Forkability doesn\'t support the language', currentLanguage + '. If you\'re keen you can open a pull request at https://github.com/basicallydan/forkability/issues');
 			}
 		}
 

--- a/spec/app.test.js
+++ b/spec/app.test.js
@@ -440,4 +440,55 @@ describe('forkability', function () {
 				done();
 			});
 	});
+
+	it('should allow uppercase spellings of languages to be used for linting', function (done) {
+		mockResponses({
+			tagsBody: ['tag1'],
+			firstCommitTreeBody: {
+				tree: [{
+					path: 'contributing.md'
+				}, {
+						path: 'readme.md'
+					}, {
+						path: 'licence.md'
+					}, {
+						path: 'codeofconduct.md'
+					}, {
+						path: 'changelog.md'
+					}, {
+						path: '.gitignore'
+					}, {
+						path: 'spec',
+						type: 'tree'
+					}, {
+						path: 'setup.py'
+					}, {
+						path: 'requirements.txt'
+					}, {
+						path: 'docs',
+						type: 'tree'
+					}, {
+						path: 'tests',
+						type: 'tree'
+					},{
+					path:'package.json'
+				}]
+			}
+		});
+
+		forkability({
+			user: 'thatoneguy',
+			repository: 'thatonerepo',
+			languages: ['Python', 'Nodejs']
+		},
+			function (err, report) {
+				should(err).eql(null);
+				console.log(report);
+				
+				report.passes.should.have.a.lengthOf(14);
+				report.failures.should.have.a.lengthOf(0);
+				report.badge.type.should.equal(forkability.badgeTypes.ok);
+				done();
+			});
+	});
 });

